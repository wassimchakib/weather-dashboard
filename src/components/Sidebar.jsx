import { useEffect, useState } from 'react';

const Sidebar = () => {
  const [width, setwidth] = useState(window.innerWidth);

  const handleWindowSizeChange = () => {
    setwidth(window.innerWidth);
  };

  useEffect(() => {
    window.addEventListener('resize', handleWindowSizeChange);
    return () => {
      window.removeEventListener('resize', handleWindowSizeChange);
    };
  }, []);

  return (
    <nav>
      <h1>Weather app</h1>
      <ul>
        <li>{width}</li>
      </ul>
    </nav>
  );
};

export default Sidebar;
