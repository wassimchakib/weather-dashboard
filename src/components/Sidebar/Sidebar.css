.sidebar {
  height: 100vh;
  background: linear-gradient(180deg, rgba(25, 27, 31, 1) 0%, rgba(62, 205, 224, 0.6) 100%);
  width: 100px;
  margin: 20px;
  border-radius: 50px;
  position: fixed;
  margin-left: 10px;
}

.sidebar__title {
  text-align: center;
  text-transform: capitalize;
  border-bottom: 1px solid var(--white);
  line-height: 1.2;
  font-weight: 400;
  margin-top: 20px;
}

.sidebar__icon {
  width: 50%;
  text-align: center;
}

.sidebar__nav {
  display: flex;
  height: 80%;
  align-items: center;
  flex-direction: column;
  justify-content: center;
}

.sidebar__nav li {
  font-size: 25px;
  width: 100%;
  margin: 10px 0;
  text-align: center;
}

.sidebar__nav li a {
  padding: 10px 0;
}

.sidebar__nav li:hover {
  background: linear-gradient(90deg, rgba(252, 252, 253, 0.25) 0%, transparent 100%);
}

.sidebar__nav li .active {
  border-left: 1px solid var(--white);
  background: linear-gradient(90deg, rgba(252, 252, 253, 0.25) 0%, transparent 100%);
}

.sidebar__nav li span {
  font-size: 18px;
  padding-left: 10px;
}

@media screen and (min-width: 768px) {
  .sidebar {
    width: 200px;
  }

  .sidebar__nav li {
    text-align: start;
    display: flex;
    align-items: center;
  }

  .sidebar__nav li a {
    padding-left: 20px;
  }

  .sidebar__title {
    display: flex;
    align-items: center;
    justify-content: center;
  }

  .sidebar__icon {
    width: 20%;
  }
}
