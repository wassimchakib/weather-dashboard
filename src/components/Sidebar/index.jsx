import { NavLink } from 'react-router-dom';
import { useEffect, useState } from 'react';
import { FontAwesomeIcon } from '@fortawesome/react-fontawesome';
import {
  faTemperatureFull,
  faUser,
  faWind,
} from '@fortawesome/free-solid-svg-icons';
import './Sidebar.css';
import IconImg from '../../assets/Lightning.svg';

const Sidebar = () => {
  const [width, setwidth] = useState(window.innerWidth);

  const handleWindowSizeChange = () => {
    setwidth(window.innerWidth);
  };

  useEffect(() => {
    window.addEventListener('resize', handleWindowSizeChange);
    return () => {
      window.removeEventListener('resize', handleWindowSizeChange);
    };
  }, []);

  return (
    <nav className="sidebar">
      <h1 className="sidebar__title">
        <img className="sidebar__icon" src={IconImg} alt="logo" />
        {width > 768 && <span>Weather App</span>}
      </h1>
      <ul className="sidebar__nav">
        <li>
          <NavLink to="dashboard">
            <FontAwesomeIcon icon={faTemperatureFull} />
            {width > 768 && <span>Dashboard</span>}
          </NavLink>
        </li>
        <li>
          <NavLink to="pollution">
            <FontAwesomeIcon icon={faWind} />
            {width > 768 && <span>Air Pollution</span>}
          </NavLink>
        </li>
        <li>
          <NavLink to="contact">
            <FontAwesomeIcon icon={faUser} />
            {width > 768 && <span>Contact</span>}
          </NavLink>
        </li>
      </ul>
    </nav>
  );
};

export default Sidebar;
