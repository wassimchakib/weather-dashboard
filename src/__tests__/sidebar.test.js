import { BrowserRouter as Router } from 'react-router-dom';
import { render } from '@testing-library/react';
import Sidebar from '../components/Sidebar';

jest.mock('react-redux');

describe('Test suite for sidebar component', () => {
  beforeEach(() => {
    jest.clearAllMocks();
  });

  it('should render sidebar', () => {
    const tree = render(
      <Router>
        <Sidebar />
      </Router>,
    );
    expect(tree).toMatchSnapshot();
  });
});
