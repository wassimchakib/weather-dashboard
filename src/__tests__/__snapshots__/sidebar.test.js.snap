// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test suite for sidebar component should render sidebar 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <nav
        class="sidebar"
      >
        <h1
          class="sidebar__title"
        >
          <img
            alt="logo"
            class="sidebar__icon"
            src="lightning.svg"
          />
          <span>
            Weather App
          </span>
        </h1>
        <ul
          class="sidebar__nav"
        >
          <li>
            <a
              class=""
              href="/dashboard"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-temperature-full "
                data-icon="temperature-full"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 320 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M160 64c-26.5 0-48 21.5-48 48V276.5c0 17.3-7.1 31.9-15.3 42.5C86.2 332.6 80 349.5 80 368c0 44.2 35.8 80 80 80s80-35.8 80-80c0-18.5-6.2-35.4-16.7-48.9c-8.2-10.6-15.3-25.2-15.3-42.5V112c0-26.5-21.5-48-48-48zM48 112C48 50.2 98.1 0 160 0s112 50.1 112 112V276.5c0 .1 .1 .3 .2 .6c.2 .6 .8 1.6 1.7 2.8c18.9 24.4 30.1 55 30.1 88.1c0 79.5-64.5 144-144 144S16 447.5 16 368c0-33.2 11.2-63.8 30.1-88.1c.9-1.2 1.5-2.2 1.7-2.8c.1-.3 .2-.5 .2-.6V112zM208 368c0 26.5-21.5 48-48 48s-48-21.5-48-48c0-20.9 13.4-38.7 32-45.3V112c0-8.8 7.2-16 16-16s16 7.2 16 16V322.7c18.6 6.6 32 24.4 32 45.3z"
                  fill="currentColor"
                />
              </svg>
              <span>
                Dashboard
              </span>
            </a>
          </li>
          <li>
            <a
              class=""
              href="/pollution"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-wind "
                data-icon="wind"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 512 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M288 32c0 17.7 14.3 32 32 32h32c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32 14.3-32 32s14.3 32 32 32H352c53 0 96-43 96-96s-43-96-96-96H320c-17.7 0-32 14.3-32 32zm64 352c0 17.7 14.3 32 32 32h32c53 0 96-43 96-96s-43-96-96-96H32c-17.7 0-32 14.3-32 32s14.3 32 32 32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H384c-17.7 0-32 14.3-32 32zM128 512h32c53 0 96-43 96-96s-43-96-96-96H32c-17.7 0-32 14.3-32 32s14.3 32 32 32H160c17.7 0 32 14.3 32 32s-14.3 32-32 32H128c-17.7 0-32 14.3-32 32s14.3 32 32 32z"
                  fill="currentColor"
                />
              </svg>
              <span>
                Air Pollution
              </span>
            </a>
          </li>
          <li>
            <a
              class=""
              href="/contact"
            >
              <svg
                aria-hidden="true"
                class="svg-inline--fa fa-user "
                data-icon="user"
                data-prefix="fas"
                focusable="false"
                role="img"
                viewBox="0 0 448 512"
                xmlns="http://www.w3.org/2000/svg"
              >
                <path
                  d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0S96 57.3 96 128s57.3 128 128 128zm-45.7 48C79.8 304 0 383.8 0 482.3C0 498.7 13.3 512 29.7 512H418.3c16.4 0 29.7-13.3 29.7-29.7C448 383.8 368.2 304 269.7 304H178.3z"
                  fill="currentColor"
                />
              </svg>
              <span>
                Contact
              </span>
            </a>
          </li>
        </ul>
      </nav>
    </div>
  </body>,
  "container": <div>
    <nav
      class="sidebar"
    >
      <h1
        class="sidebar__title"
      >
        <img
          alt="logo"
          class="sidebar__icon"
          src="lightning.svg"
        />
        <span>
          Weather App
        </span>
      </h1>
      <ul
        class="sidebar__nav"
      >
        <li>
          <a
            class=""
            href="/dashboard"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-temperature-full "
              data-icon="temperature-full"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 320 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M160 64c-26.5 0-48 21.5-48 48V276.5c0 17.3-7.1 31.9-15.3 42.5C86.2 332.6 80 349.5 80 368c0 44.2 35.8 80 80 80s80-35.8 80-80c0-18.5-6.2-35.4-16.7-48.9c-8.2-10.6-15.3-25.2-15.3-42.5V112c0-26.5-21.5-48-48-48zM48 112C48 50.2 98.1 0 160 0s112 50.1 112 112V276.5c0 .1 .1 .3 .2 .6c.2 .6 .8 1.6 1.7 2.8c18.9 24.4 30.1 55 30.1 88.1c0 79.5-64.5 144-144 144S16 447.5 16 368c0-33.2 11.2-63.8 30.1-88.1c.9-1.2 1.5-2.2 1.7-2.8c.1-.3 .2-.5 .2-.6V112zM208 368c0 26.5-21.5 48-48 48s-48-21.5-48-48c0-20.9 13.4-38.7 32-45.3V112c0-8.8 7.2-16 16-16s16 7.2 16 16V322.7c18.6 6.6 32 24.4 32 45.3z"
                fill="currentColor"
              />
            </svg>
            <span>
              Dashboard
            </span>
          </a>
        </li>
        <li>
          <a
            class=""
            href="/pollution"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-wind "
              data-icon="wind"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 512 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M288 32c0 17.7 14.3 32 32 32h32c17.7 0 32 14.3 32 32s-14.3 32-32 32H32c-17.7 0-32 14.3-32 32s14.3 32 32 32H352c53 0 96-43 96-96s-43-96-96-96H320c-17.7 0-32 14.3-32 32zm64 352c0 17.7 14.3 32 32 32h32c53 0 96-43 96-96s-43-96-96-96H32c-17.7 0-32 14.3-32 32s14.3 32 32 32H416c17.7 0 32 14.3 32 32s-14.3 32-32 32H384c-17.7 0-32 14.3-32 32zM128 512h32c53 0 96-43 96-96s-43-96-96-96H32c-17.7 0-32 14.3-32 32s14.3 32 32 32H160c17.7 0 32 14.3 32 32s-14.3 32-32 32H128c-17.7 0-32 14.3-32 32s14.3 32 32 32z"
                fill="currentColor"
              />
            </svg>
            <span>
              Air Pollution
            </span>
          </a>
        </li>
        <li>
          <a
            class=""
            href="/contact"
          >
            <svg
              aria-hidden="true"
              class="svg-inline--fa fa-user "
              data-icon="user"
              data-prefix="fas"
              focusable="false"
              role="img"
              viewBox="0 0 448 512"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M224 256c70.7 0 128-57.3 128-128S294.7 0 224 0S96 57.3 96 128s57.3 128 128 128zm-45.7 48C79.8 304 0 383.8 0 482.3C0 498.7 13.3 512 29.7 512H418.3c16.4 0 29.7-13.3 29.7-29.7C448 383.8 368.2 304 269.7 304H178.3z"
                fill="currentColor"
              />
            </svg>
            <span>
              Contact
            </span>
          </a>
        </li>
      </ul>
    </nav>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
